prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,552 @@
+        mockkObject(BlobAccess.BlobContainerMetadata)
+        every { BlobAccess.BlobContainerMetadata.build(any(), any()) } returns getBlobContainerMetadata()
+        mockkConstructor(DatabaseLookupTableAccess::class)
+    }
+
+    @AfterEach
+    fun afterEach() {
+        clearAllMocks()
+    }
+
+    @Test
+    fun `should send valid FHIR report only to receiver listening to full-elr`() {
+        // set up
+        val reportContents =
+            listOf(
+                validFHIRRecord1
+            ).joinToString()
+
+        val reportPair = createReportsWithLineage(reportContents)
+        val receiveReport = reportPair.first
+        val convertReport = reportPair.second
+        val queueMessage = generateQueueMessage(convertReport, reportContents, fhirSenderWithNoTransform)
+        val fhirFunctions = createFHIRFunctionsInstance()
+
+        // make sure action table has only what we put in there
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
+
+        // execute
+        fhirFunctions.doRoute(queueMessage, 1, createFHIRRouter())
+
+        // check results
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            // did the report get pushed to blob store correctly and intact?
+            val routedReports = verifyLineageAndFetchCreatedReportFiles(convertReport, receiveReport, txn, 1)
+            val reportAndBundles =
+                routedReports.map {
+                    Pair(
+                        it,
+                        BlobAccess.downloadBlobAsByteArray(it.bodyUrl, getBlobContainerMetadata())
+                    )
+                }
+
+            assertThat(reportAndBundles).transform {
+                pairs ->
+                pairs.map {
+                    it.second.toString(Charsets.UTF_8)
+                }
+            }.containsOnly(validFHIRRecord1)
+
+            // is the queue messaging what we expect it to be?
+            val expectedRouteQueueMessages = reportAndBundles.map { (report, fhirBundle) ->
+                FhirTranslateQueueMessage(
+                    report.reportId,
+                    report.bodyUrl,
+                    BlobAccess.digestToString(BlobAccess.sha256Digest(fhirBundle)),
+                    "phd.fhir-elr-no-transform",
+                    fhirSenderWithNoTransform.topic,
+                    "phd.elr2"
+                )
+            }.map {
+                it.serialize()
+            }
+
+            verify(exactly = 1) {
+                QueueAccess.sendMessage(
+                    elrTranslationQueueName,
+                    match {
+                    expectedRouteQueueMessages.contains(it)
+                }
+                )
+            }
+
+            // make sure action table has a new entry
+            checkActionTable(listOf(TaskAction.convert, TaskAction.receive, TaskAction.route))
+        }
+    }
+
+    @Test
+    fun `should respect jurisdictional filter and send message`() {
```

Comment:
Not following the naming here, this test is validating that the message does or does not get routed?